pkg/kubectl/resource_printer.go

```diff
@@ -337,20 +337,43 @@ type PrintOptions struct {
 // received from watches.
 type HumanReadablePrinter struct {
 	handlerMap map[reflect.Type]*handlerEntry
-	Options    PrintOptions
+	options    PrintOptions
 	lastType   reflect.Type
 }
 
 // NewHumanReadablePrinter creates a HumanReadablePrinter.
-func NewHumanReadablePrinter(options PrintOptions) *HumanReadablePrinter {
+func NewHumanReadablePrinter(pOptions PrintOptions) *HumanReadablePrinter {
 	printer := &HumanReadablePrinter{
 		handlerMap: make(map[reflect.Type]*handlerEntry),
-		Options:    options,
+		options:    pOptions,
 	}
 	printer.addDefaultHandlers()
 	return printer
 }
 
+// formatResourceName receives a resource kind, name, and boolean specifying
+// whether or not to update the current name to "kind/name"
+func formatResourceName(kind, name string, withKind bool) string {
+	if !withKind || kind == "" {
+		return name
+	}
+
+	return kind + "/" + name
+}
+
+// GetResourceName returns the currently set resource name
+func (h *HumanReadablePrinter) GetResourceName() string {
```

Comment:
> Talking about names, are there other parts of the code base talking about
"kind"?

Hm, I have not seen other parts use "kind" as well, but I did choose "kind" over type, as that is what shows up on a resource object:

`$ oc get pod <pod> -o yaml`
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/created-by: |
      {"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"default","name":"node-1","uid":"ab312b38-5991-11e6-818f-507b9dac96e1","apiVersion":"v1","resourceVersion":"30976"}}
    openshift.io/container.node.image.entrypoint: '["node"]'
    openshift.io/deployment-config.latest-version: "1"
    openshift.io/deployment-config.name: node
    openshift.io/deployment.name: node-1
    openshift.io/generated-by: OpenShiftNewApp
    openshift.io/scc: restricted
  creationTimestamp: 2016-08-03T21:20:23Z
  generateName: node-1-
  labels:
    app: node
    deployment: node-1
    deploymentconfig: node
  name: node-1-y3eid
  namespace: default
  resourceVersion: "50251"
  selfLink: /api/v1/namespaces/default/pods/node-1-y3eid
  uid: 16e9eb0a-59c0-11e6-97a3-507b9dac96e1
...
```